Source/Core/DiscIO/NANDImporter.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DiscIO/NANDImporter.h"
+
+#include "Common/Crypto/AES.h"
+#include "Common/MsgHandler.h"
+#include "Common/Swap.h"
+#include "DiscIO/NANDContentLoader.h"
+
+namespace DiscIO
+{
+void NANDImporter::ImportNANDBin(const std::string& path_to_bin)
+{
+  if (!ReadNANDBin(path_to_bin))
+    return;
+  FindSuper();
+  const std::string nand_root = File::GetUserPath(D_WIIROOT_IDX);
+  ProcessEntry(0, nand_root);
+  ExportKeys(nand_root);
+  ExtractCertificates(nand_root);
+
+  // We have to clear the cache so the new NAND takes effect
+  DiscIO::CNANDContentManager::Access().ClearCache();
+}
+
+bool NANDImporter::ReadNANDBin(const std::string& path_to_bin)
+{
+  File::IOFile file(path_to_bin, "rb");
+  if (file.GetSize() != NAND_BIN_SIZE)
+  {
+    PanicAlertT("This file does not look like a BootMii NAND backup. (0x%x does not equal 0x%x.)",
+                file.GetSize(), NAND_BIN_SIZE);
+    return false;
+  }
+  m_nand = std::vector<u8>(NAND_SIZE);
+  for (int i = 0; i < NAND_TOTAL_BLOCKS; i++)
+  {
+    file.ReadBytes(m_nand.data() + i * NAND_BLOCK_SIZE, NAND_BLOCK_SIZE);
```

Comment:
`&m_nand[i * NAND_BLOCK_SIZE]`, assuming pointer indexing isn't used to go one entry past the end of the vector.